akka-actor-typed/src/main/scala/akka/actor/typed/internal/BehaviorImpl.scala

```diff
@@ -23,16 +43,91 @@ import akka.actor.typed.scaladsl.{ ActorContext => SAC }
   def widened[O, I](behavior: Behavior[I], matcher: PartialFunction[O, I]): Behavior[O] =
     intercept(WidenedInterceptor(matcher))(behavior)
 
+  def same[T]: Behavior[T] = SameBehavior.unsafeCast[T]
+
+  def unhandled[T]: Behavior[T] = UnhandledBehavior.unsafeCast[T]
+
+  def stopped[T]: Behavior[T] = StoppedBehavior.unsafeCast[T]
+
+  def stopped[T](postStop: () => Unit): Behavior[T] =
+    new StoppedBehavior[T](OptionVal.Some((_: TypedActorContext[T]) => postStop()))
+
+  def empty[T]: Behavior[T] = EmptyBehavior.unsafeCast[T]
+
+  def ignore[T]: Behavior[T] = IgnoreBehavior.unsafeCast[T]
+
+  def failed[T](cause: Throwable): Behavior[T] = new FailedBehavior(cause).asInstanceOf[Behavior[T]]
+
+  val unhandledSignal: PartialFunction[(TypedActorContext[Nothing], Signal), Behavior[Nothing]] = {
+    case (_, _) => UnhandledBehavior
+  }
+
+  private object EmptyBehavior extends Behavior[Any](BehaviorTags.EmptyBehavior) {
+    override def toString = "Empty"
+  }
+
+  private object IgnoreBehavior extends Behavior[Any](BehaviorTags.IgnoreBehavior) {
+    override def toString = "Ignore"
+  }
+
+  object UnhandledBehavior extends Behavior[Nothing](BehaviorTags.UnhandledBehavior) {
+    override def toString = "Unhandled"
+  }
+
+  object SameBehavior extends Behavior[Nothing](BehaviorTags.SameBehavior) {
+    override def toString = "Same"
+  }
+
+  class FailedBehavior(val cause: Throwable) extends Behavior[Nothing](BehaviorTags.FailedBehavior) {
+    override def toString: String = s"Failed($cause)"
+  }
+
+  object StoppedBehavior extends StoppedBehavior[Nothing](OptionVal.None)
+
+  /**
+   * When the cell is stopping this behavior is used, so
+   * that PostStop can be sent to previous behavior from `finishTerminate`.
+   */
+  private[akka] sealed class StoppedBehavior[T](val postStop: OptionVal[TypedActorContext[T] => Unit])
+      extends Behavior[T](BehaviorTags.StoppedBehavior) {
+
+    def onPostStop(ctx: TypedActorContext[T]): Unit = {
+      postStop match {
+        case OptionVal.Some(callback) => callback(ctx)
+        case OptionVal.None           =>
+      }
+    }
+
+    override def toString = "Stopped" + {
+      postStop match {
+        case OptionVal.Some(callback) => s"(${LineNumbers(callback)})"
+        case _                        => "()"
+      }
+    }
+  }
```

Comment:
extra brackets on `onPostStop` and `toString` seem overkill but that's just a style pref.